### ISSUE TITLE
0.12 checklist missed issue causing syntax error in 0.12

### ISSUE THREAD

**Issue by geofffranks** [hashicorp/terraform#21486]
### Terraform Version
0.11.14

### Terraform Configuration Files

```hcl
data "aws_iam_policy_document" "my-policy" {
  statement {
    actions = [
      "ec2:AssociateAddress",
    ]
    resources = ["*"]
    condition = {
      test = "StringEquals"
      variable = "ec2:ResourceTag/MyTag"
      values = ["true"]
    }
  }
}
```

### Expected Behavior

Running the `terraform 0.12checklist` command should flag this as something needing . to be changed before migrating to 0.12, as this config fails in 0.12 with the following errors:

```
Error: Unsupported argument

  on test.tf line 109, in data "aws_iam_policy_document" "my-policy":
 109:     condition = {

An argument named "condition" is not expected here. Did you mean to define a
block of type "condition"?
```

### Actual Behavior
The checklist command said everything looked good, then when upgrading we got a syntax error on 0.12 for this

### Steps to Reproduce
1. terraform init # at v0.11.14
2. terraform 0.12checklist
3. Upgrade to terraform 0.12
4. terraform plan
5. terraform apply

### Additional Context
Based on the upgrade guide, it seems like the new syntax is valid for 0.11.14, so we were expecting that to have warned us of this issue.

Also, the underlying configs with issues were used as part of a module, not a top-level configuration, but i don't suspect that would matter in this scenario.

**Comment by jbardin** [comment]
Hi @geofffranks,

The checklist tools is not expected to verify 0.12 syntax, since it can't be parsed by that tool. Is this issue corrected by running `0.12upgrade`?  See the [upgrade guide](https://www.terraform.io/upgrade-guides/0-12.html) for the complete set of steps to take when upgrading.

**Comment by geofffranks** [comment]
Fair enough. We tried changing the syntax back to run 0.12upgrade and see if it caught it (we injected a plan/apply step on v0.12 in between 0.12checklist and 0.12upgrade), but the 0.12upgrade is now complaining about having the 0.12 syntax on the things it already changed.  Will close, since the 0.12checklist is working as intended. Thanks!
